src/shared/mkfs-util.c

```diff
@@ -508,19 +511,28 @@ int make_filesystem(
                 log_debug("Executing mkfs command: %s", strna(j));
         }
 
+        if (stat(node, &st) < 0 && errno != ENOENT)
+                return log_error_errno(errno, "Failed to stat %s: %m", node);
+
         r = safe_fork_full(
                         "(mkfs)",
                         stdio_fds,
                         /*except_fds=*/ NULL,
                         /*n_except_fds=*/ 0,
                         FORK_RESET_SIGNALS|FORK_RLIMIT_NOFILE_SAFE|FORK_DEATHSIG|FORK_LOG|FORK_WAIT|
-                        FORK_CLOSE_ALL_FDS|FORK_REARRANGE_STDIO,
+                        FORK_CLOSE_ALL_FDS|FORK_REARRANGE_STDIO|(S_ISBLK(st.st_mode) ? FORK_NEW_MOUNTNS : 0),
```

Comment:
why this conditionalization? why not just do this always?

Reply:
might be cool to have a FORK_NEW_MOUNTNS_GRACEFUL flag which is just like FORK_NEW_MOUNTNS, but if we get EPERM we will continue without the mount namespace...

but i figure this is not going to be enough, because we also need to then condition the remount on whether we succeeded with the mountns allocation or not, and i see no nice way how to get that out of safe_fork_full() in any way...